pkg/provider/infisical/provider.go

```diff
@@ -58,6 +67,53 @@ func (p *Provider) Capabilities() esv1.SecretStoreCapabilities {
 	return esv1.SecretStoreReadOnly
 }
 
+func performUniversalAuthLogin(ctx context.Context, store esv1.GenericStore, infisicalSpec *esv1.InfisicalProvider, sdkClient infisicalSdk.InfisicalClientInterface, kube kclient.Client, namespace string) error {
+	universalAuthCredentials := infisicalSpec.Auth.UniversalAuthCredentials
+	clientID, err := GetStoreSecretData(ctx, store, kube, namespace, universalAuthCredentials.ClientID)
+	if err != nil {
+		return err
+	}
+
+	clientSecret, err := GetStoreSecretData(ctx, store, kube, namespace, universalAuthCredentials.ClientSecret)
+	if err != nil {
+		return err
+	}
+
+	_, err = sdkClient.Auth().UniversalAuthLogin(clientID, clientSecret)
+	metrics.ObserveAPICall(constants.ProviderName, machineIdentityLoginViaUniversalAuth, err)
+
+	if err != nil {
+		return fmt.Errorf("failed to authenticate via universal auth %w", err)
+	}
+
+	return nil
+}
+
+func performAzureAuthLogin(ctx context.Context, store esv1.GenericStore, infisicalSpec *esv1.InfisicalProvider, sdkClient infisicalSdk.InfisicalClientInterface, kube kclient.Client, namespace string) error {
+	azureAuthCredentials := infisicalSpec.Auth.AzureAuthCredentials
+	identityID, err := GetStoreSecretData(ctx, store, kube, namespace, azureAuthCredentials.IdentityID)
+	if err != nil {
+		return err
+	}
+
+	resource := ""
+	if azureAuthCredentials.Resource.Name != "" {
+		resource, err = GetStoreSecretData(ctx, store, kube, namespace, azureAuthCredentials.Resource)
+	}
+	if err != nil {
+		return err
+	}
```

Comment:
```suggestion
	if azureAuthCredentials.Resource.Name != "" {
		resource, err = GetStoreSecretData(ctx, store, kube, namespace, azureAuthCredentials.Resource)
		if err != nil {
			return fmt.Errorf("failed to get secret data resource %w", err)
		}
	}
```

This error check is only relevant inside the if statement. There is no need for it outside.